src/compiler/checker.ts

```diff
@@ -5259,6 +5304,22 @@ namespace ts {
             target = getErasedSignature(target);
 
             let result = Ternary.True;
+            if (source.thisType || target.thisType) {
+                const s = source.thisType || anyType;
+                const t = target.thisType || anyType;
+                if (s !== voidType) {
+                    // void sources are assignable to anything.
+                    let related = compareTypes(getApparentType(t), getApparentType(s), reportErrors);

```

Comment:
Same reason I used `getWidenedType` elsewhere. I'll get rid of the calls since they are wrong.